File: frontend/src/services/businessService.js
/**
 * Frontend Business Profile Service.
 * Connects frontend views to /api/v1/business endpoints with dynamic business scoping.
 */

import { apiClient } from './apiClient';

export const businessService = {
  /**
   * Fetch current authorized business profile.
   * If businessId is null, resolves for the active session's authorized business.
   */
  async getBusiness(businessId = null) {
    const params = businessId ? { business_id: businessId } : {};
    return apiClient.get('/business', params);
  },

  /**
   * Create a new business proposal or registration.
   */
  async createBusiness(data) {
    return apiClient.post('/business', data);
  },

  /**
   * Update current authorized business profile.
   */
  async updateBusiness(data, businessId = null) {
    const params = businessId ? { business_id: businessId } : {};
    return apiClient.patch('/business', data, params);
  },
};

File: backend/app/services/business_service.py
"""Business profile management service layer."""

from typing import Optional, Sequence

from sqlalchemy.orm import Session

from backend.app.models.business import Business
from backend.app.repositories.business_repository import BusinessRepository
from backend.app.schemas.business import BusinessCreate, BusinessUpdate


class BusinessService:
    """Service coordinating business profile operations."""

    def __init__(self, db: Session):
        self.repo = BusinessRepository(db)

    def get_business(self, business_id: int) -> Optional[Business]:
        return self.repo.get_by_id(business_id)

    def list_user_businesses(self, user_id: int) -> Sequence[Business]:
        return self.repo.get_by_owner(user_id)

    def list_businesses(self, limit: int = 50) -> Sequence[Business]:
        """Return all registered business profiles."""
        return self.repo.list_all(limit=limit)

    def create_business(self, data: BusinessCreate) -> Business:
        return self.repo.create(data)

    def update_business(self, business_id: int, data: BusinessUpdate) -> Optional[Business]:
        business = self.repo.get_by_id(business_id)
        if not business:
            return None
        return self.repo.update(business, data)
